tmt/steps/prepare/artifact/providers/__init__.py

```diff
@@ -195,6 +199,201 @@ def _filter_artifacts(self, exclude_patterns: list[Pattern[str]]) -> Iterator[Ar
                 yield artifact
 
 
+class Repository:
+    """A class to represent a dnf/yum software repository."""
+
+    def __init__(self, logger: tmt.log.Logger, name: str, content: str):
+        self.logger = logger
+        self.name = name
+        self.content = content
+        self._repo_ids = self._get_repo_ids()
+
+    # The @overload decorator is used here to provide type hinting for multiple possible
+    # call signatures of the 'create' class method. This helps static type checkers like
+    # mypy understand that the method can be called in different ways (e.g., with 'url',
+    # 'file_path', or 'content'), even though Python doesn't support true function overloading.
+    # This specific overload defines the signature when creating a Repository from a URL.
+    @overload
+    @classmethod
+    def create(
+        cls,
+        logger: tmt.log.Logger,
+        *,
+        name: Optional[str] = None,
+        url: str,
+        file_path: None = None,
+        content: None = None,
+    ) -> "Repository": ...
+
+    # This @overload decorator defines the signature when creating a Repository from a file path.
+    # It specifies the types for parameters when 'file_path' is provided,
+    # and 'url' and 'content' are None.
+    @overload
+    @classmethod
+    def create(
+        cls,
+        logger: tmt.log.Logger,
+        *,
+        name: Optional[str] = None,
+        file_path: Path,
+        url: None = None,
+        content: None = None,
+    ) -> "Repository": ...
+
+    # This @overload decorator defines the signature when creating a Repository directly
+    # from a content string. It specifies the types for parameters when 'content'
+    # is provided, along with a required 'name' and 'url' and 'file_path' are None.
+    @overload
+    @classmethod
+    def create(
+        cls,
+        logger: tmt.log.Logger,
+        *,
+        name: str,
+        content: str,
+        url: None = None,
+        file_path: None = None,
+    ) -> "Repository": ...
+
+    @classmethod
+    def create(
```

Comment:
I'm not sure I see the point of `create()` with so many parameters. If I have a repository file URL, I would simply call `from_url()` to create a `Repository` instance. What is the use case of `create()`?